bot/app/__init__.py

```diff
@@ -25,60 +26,136 @@
+	In the background the user's id is checked against the db and added,
+	if there is no entry for the user yet.
+
+	Arguments:
+		message: telebot's message object
+	"""
+
 	u_id = message.from_user.id
 	name = message.from_user.first_name
 	is_bot = message.from_user.is_bot
 
 	user = db_objects.User(u_id, name, is_bot)
 
 	if not user.user_exists(u_id):
-		user.add_user(u_id, name, is_bot)
-
-	bot.send_message(chat_id=u_id, text=render_template('start.html', name=name), \
-		parse_mode='html')
+		user.add_user()
 
-@bot.message_handler(commands=['test'])
-def test(message):
-	with open(os.path.join(template_dir, 'test.html'), 'r') as f:
-		reply = f.read()
-	bot.send_message(chat_id=message.chat.id, text=reply, parse_mode='html')
+	with app.app_context():
+		bot.send_message(chat_id=u_id, text=render_template('start.html', name=name), \
+			parse_mode='html')
 
-# returns flyer and info on the next event when user sends /next
 @bot.message_handler(commands=['next'])
 def next_event(message):
-	with open(os.path.join(template_dir,  'next_event.html'), 'r') as f:
-		photo_caption = f.read()
-	with open(os.path.join(img_dir, 'next_event.jpg'), 'rb') as p:
-		picture=p.read()
-	bot.send_photo(chat_id=message.chat.id, photo=picture, caption=photo_caption, parse_mode='html')
-
-@bot.message_handler(commands=['message_to_all'])
-def push_message_to_all(message):
-	with open(os.path.join(template_dir, 'test.html'), 'r') as f:
-		reply = f.read()
-	uids = db.all_users()
-	for uid in uids:
-		print(uid[0])
-		if (uid != 123456789) and (uid != 987654321):
-			bot.send_message(chat_id=uid[0], text=reply, parse_mode='html')
+
+	"""/next message handler function
+
+	Upon sending the /next command the event's graphic and all available
+	info from the db are returned.
+	Artists are represented with hyperlinks. By pressing one of those,
+	the artist message handler is triggered
+
+	Arguments:
+		message: telebot's message object
+	"""
+
+	with app.app_context():
+		u_id = message.from_user.id
+	
+		try:
+			next_event = db_objects.get_next_event()
+		except:
+			return render_template('none.html')
+	
+		e_name = next_event.name
+		date = next_event.date
+		time = next_event.time
+		admission = next_event.admission
+		description = next_event.description
+		location = next_event.location
+		photo_id = os.path.join(img_dir, next_event.pic_id)
+		
+		artists_temp = db_objects.get_artists_event(e_name)
+		artists_playing = []
+		
+		for artist in artists_temp:
+			artists_playing.append(artist.name)
+
+		try:
+			bot.send_photo(chat_id=u_id, photo=open(photo_id, 'rb'))
+		except Exception as e:
+			logging.error(e)
+
+		try:
+			bot.send_message(chat_id=u_id, text=render_template('next_event.html', \
+			e_name=e_name, date=date, time=time, admission=admission, location=location, \
+			description=description, artists=artists_playing), parse_mode='html')
+		except Exception as e:
+			return render_template('none.html')
+			logging.error(e)
```

Comment:
`logging.error(e)` never gets called before you returned beforehand

Reply:
fixed